src/Button.js

```diff
@@ -98,7 +98,7 @@ class Button extends Component {
           toolbar: fab && fab.toolbarEnabled
         })}
       >
-        <a className={classes}>{this.renderIcon()}</a>
+        <a className={classes}>{icon}</a>
```

Comment:
what happens if `icon` is not supplied?

Reply:
It simply doesn't render the `icon`. The `a` nod will still be present in the `DOM` and the functionality are preserved.

Reply:
do we want that though? an empty `<a className="foo"></a>` tag??

Reply:
Without this empty tag the functionality is completely broken.
That's because there is a lot of code which depends on this `a` tag;
For example some `CSS` will not be applied nor the `backdrop` element (`<div class="fab-backdrop"></div>`) will be appended.
Have a look yourself [here](https://github.com/Dogfalo/materialize/blob/25b3ac858f458f8ec1d91f49b150138711c1000f/js/buttons.js#L38).

Reply:
Can we add a snapshot test as to what happens when I don't supply an `icon`?

Reply:
Fun fact, it seems that the `spec` we have now for the `FAB` don't use the `icon` `prop` already :grin:
Check it out [here](https://github.com/react-materialize/react-materialize/blob/b562c48061d8b163f5f3050c918f7dcc08fa6fe5/test/Button.spec.js#L159).

Reply:
True! good catch, we might need to do some changes here. Either set a default icon prop or not render, we can't have a `a` without any children. Thoughts?

Reply:
Yeah, I think the best solution is to add a `add` icon as a default prop

Reply:
Yes, I agree with you. Just one thing, I'm going to use `edit` instead of `add`, as shown on `materialize-css` [documentation](https://materializecss.com/floating-action-button.html).